File: algorithms/2_medium/W/00376_wiggle_subsequence/solution.py
class Solution:
    def wiggleMaxLength(self, nums: List[int]) -> int:
        n = len(nums)
        if n <= 1: return n

        cnt = 1
        direction = 0

        for i in range(1, n):
            if nums[i] > nums[i - 1] and direction != 1:
                direction = 1
                cnt += 1

            if nums[i] < nums[i - 1] and direction != -1:
                direction = -1
                cnt += 1

        return cnt

File: algorithms/2_medium/W/00376_wiggle_subsequence/solution.rs
impl Solution {
    pub fn wiggle_max_length(nums: Vec<i32>) -> i32 {
        let len = nums.len();
        if (len <= 1) { return len as i32; }

        let mut cnt = 1_i32;
        let mut dir = 0_i32;

        for i in 1..len {
            if nums[i] > nums[i - 1] && dir != 1 {
                dir = 1;
                cnt += 1;
            }

            if nums[i] < nums[i - 1] && dir != -1 {
                dir = -1;
                cnt += 1;
            }
        }

        cnt
    }
}
